src/hmatrix.jl

```diff
@@ -467,6 +470,54 @@ function compress!(H::HMatrix, comp)
     return H
 end
 
+"""
+    ancestors(H::HMatrix)
+
+Return all ancestors of `H`.
+"""
+function ancestors(H::HMatrix)
+    ancestors = []
+    W = H
+    while W !== parentnode(W)
+        W = parentnode(W)
+        push!(ancestors, W)
+    end
+    return ancestors
+end
+
+"""
+    issubmatrix(A::HMatrix, B::HMatrix)
+
+Return `true` if B is a submatrix of A.
+"""
+function issubmatrix(A::HMatrix, B::HMatrix)
+    for ancB in ancestors(B)
+        if A === ancB
+            return true
+        end
+    end
+    return false
+end
+
+function DataFlowTasks.memory_overlap(A::HMatrix, B::HMatrix)
+    # TODO: compare leaves in more efficient way.
+    if A === B
+        return true
+    elseif issubmatrix(A, B) || issubmatrix(B, A)
+        return true
+    end
+    chdA = leaves(A)
+    chdB = leaves(B)
+    for i in eachindex(chdA)
+        for j in eachindex(chdB)
+            if data(chdA[i]) === data(chdB[j])
+                return true
```

Comment:
I think for most use cases, we can simply:
1. Check if `A` and `B` have a common root
2. Check the intersection of the row and column ranges of `A` and `B`

When they don't have a common root, we can (probably?) assume they don't overlap?

Reply:
If they don't have a common root it is not nesseccery they don't overlap. For example, we can have this situation:
M = Matrix(...)
H = HMatrix(M, ...)
H1 = HMatrix(M, ...)
Then even we check intersections of H and H1 we need to compare their data anyway. But we can assume that the H and H1 are built correctly(without using one data matrix for more than one HMatrix) and then remove the loops altogether.